src/modules/simulation/gz_plugins/moving_platform_controller/MovingPlatformController.cpp

```diff
@@ -0,0 +1,197 @@
+
+	double default_mean_vel = 1.0f;
+	double v_x = default_mean_vel;
+
+	// Read v_x from environment variable.
+	{
+		const char *env_var_name = "PX4_GZ_PLATFORM_VEL";
+		const char *env_var_value = std::getenv(env_var_name);
+
+		if (env_var_value) {
+			try {
+				v_x = std::stof(env_var_value);
+
+			} catch (const std::invalid_argument &e) {
+				// These warnings will only be visible with sufficient verbosity level...
+				gzwarn << "Invalid argument: " << env_var_value << " is not a valid double for " << env_var_name << std::endl;
+				gzwarn << "Keeping default value of " << v_x << " m/s." << std::endl;
+
+			} catch (const std::out_of_range &e) {
+				gzwarn << "Out of range: " << env_var_value << " is out of range for " << env_var_name << std::endl;
+				gzwarn << "Keeping default value of " << v_x << " m/s." << std::endl;
+			}
+		}
+	}
+
+	_target_vel_sp = gz::math::Vector3d(v_x, 0., 0.);
+
+	// Advertise topic to communicate with VelocityControl plugin
+	std::string model_name = _model.Name(ecm);
+	std::string cmd_vel_topic = "/model/" + model_name + "/cmd_vel";
+	_platform_twist_pub = _node.Advertise<gz::msgs::Twist>(cmd_vel_topic);
+}
+
+void MovingPlatformController::PreUpdate(const gz::sim::UpdateInfo &_info, gz::sim::EntityComponentManager &ecm)
+{
+	updatePose(ecm);
+
+	// Initially, ramp up the velocity with constant acceleration. Otherwise
+	// we can't really simulate fast velocities as the drone just slips off
+	// the platform.
+
+	double max_accel_mag = 2.; // m/s^2
+	double dt = std::chrono::duration<double>(_info.dt).count(); // s
+
+	const gz::math::Vector3d vel_sp_err = _target_vel_sp - _current_vel_sp;
+
+	if (vel_sp_err.Length() < max_accel_mag * dt) {
+		// we are already almost there. set directly to avoid division by 0.
+		_current_vel_sp = _target_vel_sp;
+	} else {
+		// acclerate towards target velocity.
+		const gz::math::Vector3d accel = vel_sp_err.Normalized() * max_accel_mag;
+		_current_vel_sp = _current_vel_sp + accel * dt;
+	}
+
+	updateVelocityCommands(_current_vel_sp);
+
+	sendVelocityCommands();
+}
+
+void MovingPlatformController::updateVelocityCommands(const gz::math::Vector3d &velocity_setpoint)
+{
+	// Velocity and angular velocity = low pass filtered white noise + feedback term.
+
+	gz::math::Vector3d noise_v = gz::math::Vector3d(
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal()
+				     );
+
+	gz::math::Vector3d noise_w = gz::math::Vector3d(
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal(),
+					     gz::math::Rand::DblNormal()
+				     );
+
+	// Update rates for the filtered white noise.
+	// larger number here = faster movement
+	// write these in terms of time constants?
+	// https://ethz.ch/content/dam/ethz/special-interest/mavt/dynamic-systems-n-control/idsc-dam/Lectures/Signals-and-Systems/Lectures/Lecture%20Notes%209.pdf
+	const double alpha_v = 0.01;
+	const double alpha_w = 0.01;
```

Comment:
what is alpha? Could you briefly add in the comments what this means?

Reply:
Coefficients for the low-pass filters. I renamed them `filter_coef_*` and changed the comment to use that terminology as well to hopefully make it more obvious.